Core/NoOpRunner.Core.Tests/GameStateBuilderTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿using NoOpRunner.Core.Builders;
+using NoOpRunner.Core.Shapes;
+using NoOpRunner.Core.Shapes.EntityShapes;
+using NoOpRunner.Core.Shapes.GenerationStrategies;
+using NoOpRunner.Core.Shapes.StaticShapes;
+using Shouldly;
+using System;
+using System.Linq;
+using Xunit;
+
+namespace NoOpRunner.Core.Tests
+{
+    public class GameStateBuilderTests
+    {
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void GameStateBuilder_WhenMapIsUninitialized_AndShapeIsAdded_ShouldThrowException(bool initializeMap)
```

Comment:
Wouldn't add the '_' in the middle, but that's personal preference, this also works

Reply:
Done